.github/workflows/codeql-analysis.yml

```diff
@@ -39,7 +39,7 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3.3.0
```

Comment:
Do we want to be so specific about the version? v3 == v3.3.0 currently, I doubt we care much about minor/patch version bumps.

Reply:
I did not realize v3==v3.3.3
From some of my security training it always talks about locking in specific releases but also this action has no permissions since it is a public repo

Reply:
I normally agree with specific versions too, but I think for these types of actions and dependencies, the overhead of maintaining updates for them feels like it outweighs the benefits of taking updates more regularly and easily. I have no strong preference either way though :smile:

Reply:
I agree on the added overhead. @jpdahlke do you have an opinion?